pycloudlib/util.py

```diff
@@ -368,3 +370,19 @@ def get_query_param(uri: str, param: str) -> list:
     ['r134-fe06d70f']
     """
     return get_query_params(uri).get(param, [])
+
+
+def add_key_to_cloud_config(
+    public_key: str,
+    user_data: Optional[str] = None,
+):
+    """Add a public key to the cloud-config."""
+    if not user_data:
+        user_data_yaml = {"ssh_authorized_keys": [public_key]}
+    else:
+        user_data_yaml = yaml.safe_load(user_data)
+        if "ssh_authorized_keys" not in user_data_yaml:
+            user_data_yaml["ssh_authorized_keys"] = []
+        user_data_yaml["ssh_authorized_keys"].append(public_key)
+    new_data = yaml.safe_dump(user_data_yaml, width=999999999)
```

Comment:
Why the width=999..? Do we deal with truncation issues or issues with line-wrap in vSphere?

Reply:
pyyaml will "helpfully" add newlines on dump:
```
>>> import yaml; print(yaml.safe_dump(yaml.safe_load(open("/tmp/cloud-config.yaml"))))
ssh_authorized_keys:
- ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDVFCuncf2UzPNMH2qAupg0/8L3pIH6Y86lyfWa6nke/haXITT9xKsjLrau2aAGoyPBpgFah6TIqPIEaHCm8OjYzcpHQLRXbu3mCAK70E23ud8zubBiEZ3GSdvhedrNvxgo9p84UaEIPhDik30HzSiiv2oYrUmnnPFdacMXG8Q99khvSKnC15jylINFDAdzETKidDMV/p0xn5zVydwCxNzChJOb5nXzdTwXZ1YiB1vgRuGeOjZ77SwoTWkqFfDaUT4B7U88pTwOnAvQOIVd9LCQzcMUucAL4QdHK6XXtAciJUrG8I+h4xDt/JmpsQ4WbeYFukQt6pf1cpxSLWPigMhmFAygZZEft+gXkszELpMA6DBqy+VLjJ0/sNZzZvR7UwhG1n9o2OLdwxyw0Shxm0ZeFeFYNeNp6AaficHPEH+wGQuvgHuN35ZEgAw8MGLSoDdDPOxn5Py7gtz5fK4gSN+QtkaNFRIUZULoyBRDxr6SFVqnVvw8CxuCvkKfoygMXQk=
  james@cow

>>> 

```

the too many 9s ensures a single line stays a single line.

Reply:
Fine by me, but could we document this for the next person that asks?